django/thunderstore/repository/service_account.py

```diff
@@ -0,0 +1,57 @@
+import ulid2
+from django import forms
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+
+from thunderstore.repository.models import (
+    ServiceAccountMetadata,
+    UploaderIdentity,
+    UploaderIdentityMemberRole,
+)
+
+
+class CreateServiceAccountForm(forms.Form):
+    def __init__(self, user: User, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.user = user
+        self.fields["identity"] = forms.ModelChoiceField(
+            queryset=UploaderIdentity.objects.filter(members__user=user),
+        )
+
+    def clean_identity(self) -> UploaderIdentity:
+        identity = self.cleaned_data["identity"]
+        member = identity.members.get(user=self.user)
+        if member.role != UploaderIdentityMemberRole.owner:
+            raise ValidationError("Must be identity owner to create a service account")
```

Comment:
I prefer having a specific error for permissions compared to just giving a vague error about it not being one of the choices given but this could be combined into the `identity` field's `queryset`.

Reply:
I feel like this permission check should go to the `UploaderIdentity` model as a function for checking this permission. Alternatively we could start a convention of some sort on how and where permission checks should be defined. Either way, having it in the form itself is not a super good place, at the very least I'd separate the permission check part to a function outside of the form and call that.